tests/test_component.py

```diff
@@ -0,0 +1,52 @@
+import sys
+from pathlib import Path
+from unittest.mock import MagicMock
+
+import dask.dataframe as dd
+import pytest
+from fondant.component import FondantTransformComponent
+from fondant.dataset import FondantDataset
+
+
+components_path = Path(__file__).parent / "example_specs/components"
+
+
+def test_transform_kwargs(monkeypatch):
+    """Test that arguments are passed correctly to `Component.transform` method."""
+
+    class EarlyStopException(Exception):
+        """Used to stop execution early instead of mocking all later functionality"""
+
+    # Mock `Dataset.load_dataframe` so no actual data is loaded
+    def mocked_load_dataframe(self, spec):
+        return dd.from_dict({"a": [1, 2, 3]}, npartitions=1)
+
+    monkeypatch.setattr(FondantDataset, "load_dataframe", mocked_load_dataframe)
+
+    # Define paths to specs to instantiate component
+    arguments_dir = components_path / "arguments"
+    component_spec = arguments_dir / "component.yaml"
+    input_manifest = arguments_dir / "input_manifest.json"
+
+    # Implemented Component class
+    class MyComponent(FondantTransformComponent):
+
+        def transform(self, dataframe, *, flag, value):
+            assert flag == "success"
+            assert value == 1
+            raise EarlyStopException()
+
+    # Mock CLI arguments
```

Comment:
In #72 I used the `mock.patch` annotator for mocking the argparse arguments:
```
@mock.patch('argparse.ArgumentParser.parse_args',
            return_value=argparse.Namespace(input_manifest_path=".", output_manifest_path="result.parquet", metadata=metadata))
```
could be used as an alternative to `sys.argv`